packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/aws_pinpoint_user_profile.dart

```diff
@@ -7,17 +7,17 @@ import 'package:amplify_core/amplify_core.dart';
 /// Extends the category-defined UserProfile class to
 /// include features supported relevant to Pinpoint only.
 /// {@endtemplate}
-class AWSPinpointUserProfile extends AnalyticsUserProfile {
+class AWSPinpointUserProfile extends UserProfile {
   /// {@macro amplify_analytics_pinpoint_dart.aws_pinpoint_user_profile}
   AWSPinpointUserProfile({
     super.name,
     super.email,
     super.plan,
     super.location,
-    super.analyticsProperties,
+    super.customProperties,
     required this.userAttributes,
   });
 
   /// Analytics attributes for the Pinpoint User.
-  final AnalyticsProperties userAttributes;
+  final Map<String, List<String>> userAttributes;
```

Comment:
This new type is for aligning with amplify-swift correct? `userAttributes` is nullable in  amplify-swift though

Reply:
Thanks good catch.  Idea behind non nullable was since you're using the Pinpoint subtype you should be forced to actually use that field, but would prefer to be consistent with iOS here.  And underlying data representation technically doesn't require it anyway.